### ISSUE TITLE
How to I run code with WSL?

### ISSUE THREAD

**Issue by DemoCrazer** [formulahendry/vscode-code-runner#319]
I setup WSL Ubuntu but I don't know how to connect Code-Runner with WSL.
How to I run code with WSL using Code-Runner?

**Comment by formulahendry** [comment]
@garytyler Did you set WSL for the VS Code terminal? If yes, the Code Runner would automatically change the file path.

**Comment by garytyler** [comment]
@formulahendry Yes, I have. I will have a look at #106 and see if I missed anything. Thanks!

**Comment by garytyler** [comment]
@formulahenry, sorry for the delay closing this. It’s working well now. Thanks.

**Comment by garytyler** [comment]
@formulahendry, sorry for the delay in closing this. It’s working well now.

**Comment by garytyler** [comment]
@formulahendry, sorry for the delay in closing this. It’s working well now.

**Comment by garytyler** [comment]
@formulahendry, sorry for the delay. It’s working well now.

**Comment by formulahendry** [comment]
Good to know it is working.

**Comment by jcklpe** [comment]
@garytyler  How did you get it to work? I already have bash set up as my default shell in vscode settings but code runner still tries to run a windows version of node for me

**Comment by jcklpe** [comment]
Nvm got it working. Just had to also add:

"code-runner.runInTerminal": true 

in addition to 

"code-runner.terminalRoot": "/mnt/",

**Comment by JamesQian1999** [comment]
I also setup WSL Ubuntu but I can't run with WSL in right the path.
How to I run code with WSL in the right path?

This is my settings.json
`{
    "code-runner.runInTerminal": true,
    "terminal.integrated.shell.windows": "C:\\WINDOWS\\System32\\wsl.exe",
    "code-runner.executorMap": {
    "c": "cd $(wslpath $dir) & gcc $fileName -o $fileNameWithoutExt & ./$fileNameWithoutExt",
    "cpp": "g++ $fileName -o $fileNameWithoutExt -std=c++11 & ./$fileNameWithoutExt"},
    "python": "python3 -u ./$fileName",
}`

When I run my code with "ctrl+alt+N"
The output became:
`jamesqian@J:/mnt/c/Users/Jamesqian/Desktop/tmp$ cd $(wslpath "c:\Users\Jamesqian\Desktop\tmp\") & gcc test.c -o test & ./test`

wslpath is [https://github.com/laurent22/wslpath](url)

**Comment by MauriAndresMU1313** [comment]
@JamesQian1999 Hi, you could share your step-by-step? Is there a way to do that without using wslpath? Do you know any nice tutorial about this on youtube? Thank

**Comment by Sbrjt** [comment]
TLDR: 

Go to command palette (Ctrl + Shift + P), search for default terminal and select WSL.

![Image](https://github.com/user-attachments/assets/3883df99-cd6c-4e73-ab50-a5d85a3bd504)

Go to code-runner settings and add `/mnt/` to the terminal path.

![Image](https://github.com/user-attachments/assets/3de7eb5f-03c1-406f-bb27-fdd52c7c30a1)
